### PR TITLE
Fixing PerformanceTest: table_exists condition, move getMemoryAmount function to file. Add more debug helpers

### PR DESCRIPTION
**Pull request by proller**
table_exists condition, move getMemoryAmount function to file. Add more debug helpers

### CODE REVIEW

File: dbms/src/Common/getMemoryAmount.h [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#pragma once
+#include <cstddef>
+
+namespace DB {
+
+size_t getMemoryAmount();
```
Comment (alexey-milovidov) [inline dbms/src/Common/getMemoryAmount.h]: No comments. No comment that function returns zero on unsupported platform.

File: dbms/src/Common/getMemoryAmount.h [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#pragma once
+#include <cstddef>
+
+namespace DB {
```
Comment (alexey-milovidov) [inline dbms/src/Common/getMemoryAmount.h]: Style.

File: dbms/src/Common/getMemoryAmount.cpp [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+namespace DB
+{
+
+size_t getMemoryAmount()
+{
+#if __has_include(<sys/sysinfo.h>)
+    struct sysinfo system_information;
+    if (sysinfo(&system_information))
+        return 0;
```
Comment (alexey-milovidov) [inline dbms/src/Common/getMemoryAmount.cpp]: You should throw an exception.

File: dbms/src/Common/iostream_debug_helpers.cpp [hunk 4]
```diff
@@ -79,3 +79,18 @@ std::ostream & operator<<(std::ostream & stream, const DB::IColumn & what)
...
+
+std::ostream & operator<<(std::ostream & stream, const DB::Connection::Packet & what) {
+    stream << "Connection::Packet("
+        << "type = " << what.type;
+        // types description: Core/Protocol.h
+    if (what.exception)
+        stream << "exception = " << what.exception.get();
+    //TODO: profile_info
+    stream
+        << ") {"
```
Comment (alexey-milovidov) [inline dbms/src/Common/iostream_debug_helpers.cpp]: Inconsistent style of whitespaces. See above.

File: dbms/src/Server/PerformanceTest.cpp [hunk 5]
```diff
@@ -667,34 +665,21 @@ class PerformanceTest
...
                 }
 
             if (precondition == "ram_size")
             {
-#if __has_include(<sys/sysinfo.h>)
-                struct sysinfo *system_information = new struct sysinfo();
-                if (sysinfo(system_information))
+                size_t ram_size_needed = config->getUInt64("preconditions.ram_size");
+                size_t actual_ram_mb = getMemoryAmount() / 1024 / 1024;
+                if (ram_size_needed > actual_ram_mb)
```
Comment (alexey-milovidov) [inline dbms/src/Server/PerformanceTest.cpp]: Inconsistent style of sizes in configuration. All sizes should be in number of bytes.

File: dbms/src/Server/PerformanceTest.cpp [hunk 6]
```diff
@@ -667,34 +665,21 @@ class PerformanceTest
...
-                {
-                    size_t ram_size_needed = config->getUInt64("preconditions.ram_size");
-                    size_t actual_ram = system_information->totalram / 1024 / 1024;
-                    if (ram_size_needed > actual_ram)
-                    {
-                        std::cerr << "Not enough RAM" << std::endl;
-                        delete system_information;
-                        return false;
-                    }
+                    std::cerr << "Not enough RAM: need=" << ram_size_needed << " present=" << actual_ram_mb << std::endl;
```
Comment (alexey-milovidov) [inline dbms/src/Server/PerformanceTest.cpp]: No whitespaces around `=`.
No units.

File: dbms/src/Server/PerformanceTest.cpp [hunk 7]
```diff
@@ -711,11 +696,15 @@ class PerformanceTest
...
+                    if (packet.type == Protocol::Server::Data)
+                    {
                         for (const ColumnWithTypeAndName & column : packet.block.getColumns())
                         {
-                            if (column.name == "result" && column.column->getDataAt(0).data != nullptr) {
-                                exist = column.column->get64(0);
+                            if (column.name == "result" && column.column->size() > 0)
+                            {
+                                Field field;
+                                column.column->get(0, field);
```
Comment (alexey-milovidov) [inline dbms/src/Server/PerformanceTest.cpp]: So, why previous code was incorrect?

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]

File: dbms/src/Server/PerformanceTest.cpp (hunk 7 above)
Comment (proller) [inline dbms/src/Server/PerformanceTest.cpp]: it was fixed here - https://github.com/yandex/ClickHouse/commit/8c9aa37318c87cb0d226721e74fc450d50b03fb3

File: dbms/src/Common/getMemoryAmount.h [hunk 8]
```diff
@@ -0,0 +1,11 @@
+#pragma once
+#include <cstdint>
+
+namespace DB
+{
+/**
+* Returns the size of physical memory (RAM) in bytes.
+* Returns 0 on unsupported platform
+*/
+int64_t getMemoryAmount();
```
Comment (alexey-milovidov) [inline dbms/src/Common/getMemoryAmount.h]: Why not unsigned?

File: dbms/src/Server/PerformanceTest.cpp [hunk 9]
```diff
@@ -662,34 +660,24 @@ class PerformanceTest
...
             {
-#if __has_include(<sys/sysinfo.h>)
-                struct sysinfo *system_information = new struct sysinfo();
-                if (sysinfo(system_information))
-                {
-                    std::cerr << "Failed to check system RAM size" << std::endl;
-                    delete system_information;
-                }
-                else
+                size_t ram_size_needed = config->getUInt64("preconditions.ram_size");
```
Comment (alexey-milovidov) [inline dbms/src/Server/PerformanceTest.cpp]: RAM size should be configured in bytes.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]

File: dbms/src/Server/PerformanceTest.cpp [hunk 10]
```diff
@@ -662,34 +660,24 @@ class PerformanceTest
...
                 {
-                    size_t ram_size_needed = config->getUInt64("preconditions.ram_size");
-                    size_t actual_ram = system_information->totalram / 1024 / 1024;
-                    if (ram_size_needed > actual_ram)
-                    {
-                        std::cerr << "Not enough RAM" << std::endl;
-                        delete system_information;
-                        return false;
-                    }
+                    std::cerr << "Not enough RAM: need =" << ram_size_needed << " present=" << actual_ram << std::endl;
```
Comment (alexey-milovidov) [inline dbms/src/Server/PerformanceTest.cpp]: Inconsistent whitespaces.

**Pull request merged**